b1ddi/resource_dns_record.go

```diff
@@ -365,6 +366,15 @@ func resourceDataRecord() *schema.Resource {
 func resourceDataRecordCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*b1ddiclient.Client)
 
+	opts, err := updateDataRecordOptions(d.Get("options"), d.Get("type").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	rData, err := updateDataRecordRData(d.Get("rdata"), d.Get("type").(string))
```

Comment:
I tried implementing the `flatten` function. Since the response from our API returns the correct datatype, the inverse operation really isn't needed. What I noticed is that, terraform converts the API response values to a string representation while rendering it.